lib/charms/kafka/v0/kafka_snap.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""## Overview
+
+`kafka_snap.py` provides a collection of common functions for managing snap installation and
+config parsing common to both the Kafka and ZooKeeper charms
+
+"""
+
+import subprocess
+import logging
+from typing import Dict, List
+
+from ops.model import StatusBase
+from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v1 import snap
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
+
+logger = logging.getLogger(__name__)
+
+# The unique Charmhub library identifier, never change it
+LIBID = "73d0f23286dd469596d358905406dcab"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 3
+
+
+class KafkaSnap:
+    def __init__(self) -> None:
+        self.default_config_path = "/snap/kafka/current/opt/kafka/config/"
+        self.snap_config_path = "/var/snap/kafka/common/"
+        self.kafka = None
+
+    def install_kafka_snap(self) -> StatusBase:
+        """Loads the Kafka snap from LP, returning a StatusBase for the Charm to set."""
+
+        try:
+            apt.update()
+            apt.add_package("snapd")
+            cache = snap.SnapCache()
+            kafka = cache["kafka"]
+
+            if not kafka.present:
+                kafka.ensure(snap.SnapState.Latest, channel="rock/edge")
+
+            self.kafka = kafka 
+            return MaintenanceStatus("sucessfully installed kafka snap")
```

Comment:
@marcoppenheimer , I had some chats in others PRs about status setting outside the charm code.
My felling is that all status manipulation should be done exclusively within the charm code, in order to preserve single responsibility principle.
In that way, libraries (and any other helpers to that matter), would only return custom errors that can be parsed on the charm code, where treatment is due to happen.
So a kafka-snap library would facilitate and abstract kafka-snap package management, service management, etc, but not charm lifecycle.
Ping me if you want to expand on that.